x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_url_state.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useReducer } from 'react';
+import { TimeRange } from '@kbn/es-query';
+import DateMath from '@kbn/datemath';
+import deepEqual from 'fast-deep-equal';
+import * as rt from 'io-ts';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { constant, identity } from 'fp-ts/lib/function';
+import { enumeration } from '@kbn/securitysolution-io-ts-types';
+import { FilterStateStore } from '@kbn/es-query';
+import { useUrlState } from '../../../../utils/use_url_state';
+import { useKibanaTimefilterTime } from '../../../../hooks/use_kibana_timefilter_time';
+
+const DEFAULT_FROM_MINUTES_VALUE = 15;
+
+const INITIAL_DATE = new Date();
+export const INITIAL_DATE_RANGE = { from: 'now-15m', to: 'now' };
+export const DEFAULT_QUERY = {
+  language: 'kuery',
+  query: '',
+};
+
+// with this approach, all state will be in one variable in the url
```

Comment:
I guess this comment can be removed. Wdyt?

Reply:
I agree, removed 👍